File: 29_diff_two_arrays.js
// Difference between two arrays

function diffArray(arr1, arr2){
    var result = [];

    for (var i=0; i<arr1.length; i++){
        if(arr2.indexOf(arr1[i]) === -1){
            result.push(arr1[i]);
        }
    }

    for (var j=0; j<arr2.length; j++){
        if(arr1.indexOf(arr2[j]) === -1){
            result.push(arr2[j]);
        }
    }

    return result;
}

// using filter function

function diffArrayFilter(arr1, arr2){
    var combo = arr1.concat(arr2);

    return combo.filter(function(num){
        if(arr1.indexOf(num) === -1 || arr2.indexOf(num) === -1)
        {
            return num;
        }
    });
}

console.log(diffArray([1,2,3,5],[1,2,3,4,5]))
console.log(diffArray([1,2,3],[4,3,6]))

console.log(diffArrayFilter([1,2,3,5],[1,2,3,4,5]))
console.log(diffArrayFilter([1,2,3],[4,3,6]))

File: 29_diff_two_arrays.py
# Difference bwtween two arrays

def diffArray(arr1, arr2):
    result = []

    for i in range(len(arr1)):
        if not arr1[i] in arr2:
            result.append(arr1[i])

    for j in range(len(arr2)):
        if not arr2[j] in arr1:
            result.append(arr2[j])

    return result

# using list comprehensions
def diffArrayList(arr1, arr2):
    combo = arr1 + arr2
    return [x for x in combo if combo.count(x) == 1]



print(diffArray([1,2,3,5],[1,2,3,4,5]))
print(diffArray([1,2,3],[4,3,6]))
print(diffArrayList([1,2,3,5],[1,2,3,4,5]))
print(diffArrayList([1,2,3],[4,3,6]))

